bpf/lib/egress_gateway.h

```diff
@@ -0,0 +1,91 @@
+/* SPDX-License-Identifier: GPL-2.0 */
+/* Copyright (C) 2021 Authors of Cilium */
+
+#ifndef __EGRESS_GATEWAY_H_
+#define __EGRESS_GATEWAY_H_
+
+#include <bpf/compiler.h>
+#include <bpf/ctx/ctx.h>
+
+#include <linux/ip.h>
+
+#include "maps.h"
+
+#if defined(ENABLE_EGRESS_GATEWAY)
```

Comment:
Nit: I think the preference is on `#ifdef` unless we can't do without `defined`. IIRC, Quentin even translated lots of `#if defined` to `#ifdef` a while back.